File: VB/AttendanceRepository.Report.vb
Imports Attendance.AttendanceBusiness
Imports Framework.UI

Partial Class AttendanceRepository
    Inherits AttendanceRepositoryBase 
    Public Function GET_REPORT() As DataTable
        Dim dt As DataTable
        Using rep As New AttendanceBusinessClient
            Try
                dt = rep.GET_REPORT()
                Return dt
            Catch ex As Exception
                rep.Abort()
                Throw ex
            End Try
        End Using
    End Function

    Public Function GetReportById(ByVal _filter As Se_ReportDTO, ByVal PageIndex As Integer,
                                       ByVal PageSize As Integer,
                                       ByRef Total As Integer,
                                       Optional ByVal Sorts As String = "CODE ASC") As List(Of Se_ReportDTO)

        Dim lstTitle As List(Of Se_ReportDTO)

        Using rep As New AttendanceBusinessClient
            Try
                lstTitle = rep.GetReportById(_filter, PageIndex, PageSize, Total, Me.Log, Sorts)
                Return lstTitle
            Catch ex As Exception
                rep.Abort()
                Throw ex
            End Try
        End Using

        Return Nothing
    End Function

    Public Function GETORGNAME(ByVal obj As ParamDTO) As DataTable
        Dim dt As DataTable
        Using rep As New AttendanceBusinessClient
            Try
                dt = rep.GETORGNAME(obj, Me.Log)
                Return dt
            Catch ex As Exception
                rep.Abort()
                Throw ex
            End Try
        End Using
    End Function

    Public Function GET_AT001(ByVal obj As ParamDTO) As DataSet
        Dim dt As DataSet
        Using rep As New AttendanceBusinessClient
            Try
                dt = rep.GET_AT001(obj, Me.Log)
                Return dt
            Catch ex As Exception
                rep.Abort()
                Throw ex
            End Try
        End Using
    End Function

    Public Function GET_AT002(ByVal obj As ParamDTO) As DataSet
        Dim dt As DataSet
        Using rep As New AttendanceBusinessClient
            Try
                dt = rep.GET_AT002(obj, Me.Log)
                Return dt
            Catch ex As Exception
                rep.Abort()
                Throw ex
            End Try
        End Using
    End Function

    Public Function GET_AT003(ByVal obj As ParamDTO) As DataSet
        Dim dt As DataSet
        Using rep As New AttendanceBusinessClient
            Try
                dt = rep.GET_AT003(obj, Me.Log)
                Return dt
            Catch ex As Exception
                rep.Abort()
                Throw ex
            End Try
        End Using
    End Function

    Public Function GET_AT004(ByVal obj As ParamDTO) As DataSet
        Dim dt As DataSet
        Using rep As New AttendanceBusinessClient
            Try
                dt = rep.GET_AT004(obj, Me.Log)
                Return dt
            Catch ex As Exception
                rep.Abort()
                Throw ex
            End Try
        End Using
    End Function

    Public Function GET_AT005(ByVal obj As ParamDTO) As DataSet
        Dim dt As DataSet
        Using rep As New AttendanceBusinessClient
            Try
                dt = rep.GET_AT005(obj, Me.Log)
                Return dt
            Catch ex As Exception
                rep.Abort()
                Throw ex
            End Try
        End Using
    End Function

    Public Function GET_AT006(ByVal obj As ParamDTO) As DataSet
        Dim dt As DataSet
        Using rep As New AttendanceBusinessClient
            Try
                dt = rep.GET_AT006(obj, Me.Log)
                Return dt
            Catch ex As Exception
                rep.Abort()
                Throw ex
            End Try
        End Using
    End Function

    Public Function GET_AT007(ByVal obj As ParamDTO) As DataSet
        Dim dt As DataSet
        Using rep As New AttendanceBusinessClient
            Try
                dt = rep.GET_AT007(obj, Me.Log)
                Return dt
            Catch ex As Exception
                rep.Abort()
                Throw ex
            End Try
        End Using
    End Function

    Public Function ExportReport(ByVal _reportCode As String, ByVal _pkgName As String, ByVal obj As ParamDTO) As DataSet
        Dim dt As DataSet
        Using rep As New AttendanceBusinessClient
            Try
                dt = rep.ExportReport(_reportCode, _pkgName, obj, Me.Log)
                Return dt
            Catch ex As Exception
                rep.Abort()
                Throw ex
            End Try
        End Using
    End Function

End Class

File: C#/AttendanceRepository.Report.cs
using System;
using Attendance.AttendanceBusiness;
using Framework.UI;

partial class AttendanceRepository : AttendanceRepositoryBase
{
    public DataTable GET_REPORT()
    {
        DataTable dt;
        using (AttendanceBusinessClient rep = new AttendanceBusinessClient())
        {
            try
            {
                dt = rep.GET_REPORT();
                return dt;
            }
            catch (Exception ex)
            {
                rep.Abort();
                throw ex;
            }
        }
    }

    public List<Se_ReportDTO> GetReportById(Se_ReportDTO _filter, int PageIndex, int PageSize, ref int Total, string Sorts = "CODE ASC")
    {
        List<Se_ReportDTO> lstTitle;

        using (AttendanceBusinessClient rep = new AttendanceBusinessClient())
        {
            try
            {
                lstTitle = rep.GetReportById(_filter, PageIndex, PageSize, Total, this.Log, Sorts);
                return lstTitle;
            }
            catch (Exception ex)
            {
                rep.Abort();
                throw ex;
            }
        }

        return null;
    }

    public DataTable GETORGNAME(ParamDTO obj)
    {
        DataTable dt;
        using (AttendanceBusinessClient rep = new AttendanceBusinessClient())
        {
            try
            {
                dt = rep.GETORGNAME(obj, this.Log);
                return dt;
            }
            catch (Exception ex)
            {
                rep.Abort();
                throw ex;
            }
        }
    }

    public DataSet GET_AT001(ParamDTO obj)
    {
        DataSet dt;
        using (AttendanceBusinessClient rep = new AttendanceBusinessClient())
        {
            try
            {
                dt = rep.GET_AT001(obj, this.Log);
                return dt;
            }
            catch (Exception ex)
            {
                rep.Abort();
                throw ex;
            }
        }
    }

    public DataSet GET_AT002(ParamDTO obj)
    {
        DataSet dt;
        using (AttendanceBusinessClient rep = new AttendanceBusinessClient())
        {
            try
            {
                dt = rep.GET_AT002(obj, this.Log);
                return dt;
            }
            catch (Exception ex)
            {
                rep.Abort();
                throw ex;
            }
        }
    }

    public DataSet GET_AT003(ParamDTO obj)
    {
        DataSet dt;
        using (AttendanceBusinessClient rep = new AttendanceBusinessClient())
        {
            try
            {
                dt = rep.GET_AT003(obj, this.Log);
                return dt;
            }
            catch (Exception ex)
            {
                rep.Abort();
                throw ex;
            }
        }
    }

    public DataSet GET_AT004(ParamDTO obj)
    {
        DataSet dt;
        using (AttendanceBusinessClient rep = new AttendanceBusinessClient())
        {
            try
            {
                dt = rep.GET_AT004(obj, this.Log);
                return dt;
            }
            catch (Exception ex)
            {
                rep.Abort();
                throw ex;
            }
        }
    }

    public DataSet GET_AT005(ParamDTO obj)
    {
        DataSet dt;
        using (AttendanceBusinessClient rep = new AttendanceBusinessClient())
        {
            try
            {
                dt = rep.GET_AT005(obj, this.Log);
                return dt;
            }
            catch (Exception ex)
            {
                rep.Abort();
                throw ex;
            }
        }
    }

    public DataSet GET_AT006(ParamDTO obj)
    {
        DataSet dt;
        using (AttendanceBusinessClient rep = new AttendanceBusinessClient())
        {
            try
            {
                dt = rep.GET_AT006(obj, this.Log);
                return dt;
            }
            catch (Exception ex)
            {
                rep.Abort();
                throw ex;
            }
        }
    }

    public DataSet GET_AT007(ParamDTO obj)
    {
        DataSet dt;
        using (AttendanceBusinessClient rep = new AttendanceBusinessClient())
        {
            try
            {
                dt = rep.GET_AT007(obj, this.Log);
                return dt;
            }
            catch (Exception ex)
            {
                rep.Abort();
                throw ex;
            }
        }
    }

    public DataSet ExportReport(string _reportCode, string _pkgName, ParamDTO obj)
    {
        DataSet dt;
        using (AttendanceBusinessClient rep = new AttendanceBusinessClient())
        {
            try
            {
                dt = rep.ExportReport(_reportCode, _pkgName, obj, this.Log);
                return dt;
            }
            catch (Exception ex)
            {
                rep.Abort();
                throw ex;
            }
        }
    }
}
